components/sup/src/event/nats_message_stream.rs

```diff
@@ -40,57 +43,56 @@ impl NatsMessageStream {
                                 server_certificate,
                                 .. } = config;
 
-        let mut client = Client::new(vec![url]);
-
-        // Configure the client connect message
-        client.connect_mut()
-              .await
-              .name(format!("hab_client_{}", supervisor_id))
-              .verbose(true)
-              .token(token.to_string());
-
-        // Configure the tls connector
-        let mut tls_connector = TlsConnector::builder();
-        for certificate in habitat_core::tls::native_tls_wrapper::certificates(None)? {
-            tls_connector.add_root_certificate(certificate);
-        }
-        if let Some(certificate) = server_certificate {
-            tls_connector.add_root_certificate(certificate.into());
+        let name = format!("hab_client_{}", supervisor_id); 
+ 
+        //  Not sure how to add a vector into the nats_options object below (in Options::with_token(...))
+        //  when the interface is add_root_certificate
+        let mut hab_certs: Vec<PathBuf> = Vec::new();
+        for certificate in habitat_core::tls::native_tls_wrapper::certificates_pathbuf(None)? {
+            hab_certs.push(certificate);
         }
-        let tls_connector = tls_connector.build()?;
-        client.set_tls_config(tls_connector).await;
+
+        let nats_options  = match server_certificate {
```

Comment:
make this `let mut nats_options  = match server_certificate {` so that the `nats_options` are mutable and can be reassigned to the same variable as you add certs.

Reply:
That worked - thanks!